src/main/cpp/src/cast_string.cu

```diff
@@ -271,4 +756,37 @@ std::unique_ptr<column> string_to_integer(data_type dtype,
     dtype, detail::string_to_integer_impl{}, string_col, ansi_mode, stream, mr);
 }
 
+/**
+ * @brief Convert a string column into an decimal column.
+ *
+ * @param precision precision of input data
+ * @param scale scale of input data
+ * @param string_col Incoming string column to convert to decimals.
+ * @param ansi_mode If true, strict conversion and throws on erorr.
+ *                  If false, null invalid entries.
+ * @param stream Stream on which to operate.
+ * @param mr Memory resource for returned column
+ * @return std::unique_ptr<column> Decimal column that was created from string_col.
+ */
+std::unique_ptr<column> string_to_decimal(int32_t precision,
+                                          int32_t scale,
+                                          strings_column_view const& string_col,
+                                          bool ansi_mode,
+                                          rmm::cuda_stream_view stream,
+                                          rmm::mr::device_memory_resource* mr)
+{
+  data_type dtype = [precision, scale]() {
+    if (precision <= cuda::std::numeric_limits<int32_t>::digits10)
+      return data_type(type_id::DECIMAL32, -scale);
```

Comment:
nit: can we have the scale passed in be the CUDF scale and not the java/scala/spark scale?  This is for consistency. All of the other CUDF java APIs take the CUDF scale.

Reply:
This is actually helpful for understanding the code as well. I'll make this change.